src/olympia/api/urls.py

```diff
@@ -1,11 +1,50 @@
 from django.urls import include, re_path
 
+from drf_yasg import openapi
+from drf_yasg.views import get_schema_view
+from rest_framework import permissions
+
 from olympia.accounts.urls import accounts_v3, accounts_v4, auth_urls
 from olympia.addons.api_urls import addons_v3, addons_v4, addons_v5
 from olympia.amo.urls import api_patterns as amo_api_patterns
 from olympia.ratings.api_urls import ratings_v3, ratings_v4
 
 
+def get_versioned_api_routes(version, url_patterns):
+    route_pattern = r'^{}/'.format(version)
+
+    schema_view = get_schema_view(
+        openapi.Info(
+            title='AMO API',
+            default_version=version,
+            description='The official API for addons.mozilla.org.',
+        ),
+        public=True,
+        permission_classes=(permissions.AllowAny,),
+    )
+
+    routes = [
+        re_path(
+            r'^swagger(?P<format>\.json|\.yaml)$',
+            schema_view.without_ui(cache_timeout=0),
+            name='schema-json',
+        ),
+        re_path(
+            r'^swagger/$',
+            schema_view.with_ui('swagger', cache_timeout=0),
+            name='schema-swagger-ui',
+        ),
+        re_path(
+            r'^redoc/$',
+            schema_view.with_ui('redoc', cache_timeout=0),
+            name='schema-redoc',
+        ),
```

Comment:
Maybe we should do this on local envs/dev only for now ? This isn't completely ready/useful for production for all APIs...

Reply:
Added settings.DEBUG for now for simplicity.